views/tables/ConfigurationsTable/ConfigurationsTable.test.jsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react'
+import { render, screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import ConfigurationsTable from '.'
+import { createUser, createConfiguration } from '../../../test/fixtures'
+
+describe('ConfigurationsTable', () => {
+  const user = createUser({ uid: 'user', preferences: { config: '1' } })
+  const configurations = [
+    createConfiguration({
+      name: 'config1',
+      owner: user.uid,
+      readers: ['user', 'foo', 'bar'],
+    }),
+    createConfiguration({
+      _id: '2',
+      name: 'config2',
+      owner: 'foo',
+      readers: ['foo'],
+    }),
+    createConfiguration({
+      _id: '3',
+      name: 'config3',
+      readers: ['user', 'bar'],
+      owner: 'bar',
+    }),
+  ]
+  const defaultProps = {
+    configurations,
+    onDefaultChange: () => {},
+    onDelete: () => {},
+    onDuplicate: () => {},
+    onEdit: () => {},
+    onShare: () => {},
+    user,
+  }
+  const elements = {
+    editBtn: () => screen.getByTestId('EditIcon'),
+    deleteBtn: () => screen.getByTestId('DeleteIcon'),
+    duplicateBtn: () => screen.getByTestId('ContentCopyIcon'),
+    shareBtn: () => screen.getByTestId('ShareIcon'),
+    setDefaultBtn: () => screen.getByTestId('SettingsApplicationsIcon'),
+    tableRow: (tableRow) => screen.getByTestId(`row-${tableRow}`),
+  }
+  const renderConfigurationsTable = (props = defaultProps) =>
+    render(<ConfigurationsTable {...props} />)
+  const actions = {
+    openTableRowMenu: async (tableRow) => {
+      return await userEvent.click(
+        screen.getByRole('button', {
+          name: `open menu for ${configurations[tableRow].name}`,
+        })
+      )
+    },
+    deleteConfiguration: async (tableRow) => {
+      await actions.openTableRowMenu(tableRow)
+      await userEvent.click(elements.deleteBtn())
+    },
+    duplicateConfiguration: async (tableRow) => {
+      await actions.openTableRowMenu(tableRow)
+      await userEvent.click(elements.duplicateBtn())
+    },
+    shareConfiguration: async (tableRow) => {
+      await actions.openTableRowMenu(tableRow)
+      await userEvent.click(elements.shareBtn())
+    },
+    setDefaultConfiguration: async (tableRow) => {
+      await actions.openTableRowMenu(tableRow)
+      await userEvent.click(elements.setDefaultBtn())
+    },
+  }
+
+  test('renders chart', () => {
```

Comment:
we're mixing up `test` and `it` in this file. Let's choose one or the other.

Reply:
[47a7352](https://github.com/AMP-SCZ/dpdash/pull/487/commits/47a7352aa90256c68d3515ee70b8a513b3052989)